pkgs/tools/misc/aoc-cli/default.nix

```diff
@@ -0,0 +1,31 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, pkg-config
+, openssl
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "aoc-cli";
+  version = "0.6.0";
+
+  src = fetchFromGitHub {
+    owner = "scarvalhojr";
+    repo = pname;
+    rev = version;
+    hash = "sha256-aDbW//DmJiQO0/RJ1JnxAm4ImsfO6TYajmlO43CruWo=";
+  };
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [ openssl ];
```

Comment:
```suggestion
  buildInputs = [ openssl ] ++ lib.optionals stdenv.isDarwin [
    darwin.apple_sdk.frameworks.Security
  ];
```

Reply:
Fixed it, let me know if that works. Is there a command to build it for darwin on my linux system to test?

Reply:
I just use github actions like this: https://github.com/figsoda/nixpkgs-test